timbermill-java/timbermill-local/src/main/java/com/datorama/oss/timbermill/unit/Task.java

```diff
@@ -95,9 +97,12 @@ public Task(List<Event> events, String index, long daysRotation, String timbermi
 				setEndTime(endTime);
 			}
 
-			ZonedDateTime dateToDelete = e.getDateToDelete(daysRotation);
-			if (dateToDelete != null) {
-				this.setDateToDelete(dateToDelete);
+			if (!hasStart) {
```

Comment:
You don't know that the first event will also be the start event

Reply:
please add a test for this. multiple events in the same batch but opposite order, to make sure we get the datetodelete of the start event.